public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -181,49 +211,54 @@ const Policy: FC<PolicyComponentProps> = ({
       </Fragment>
     );
   }
-
   // TODO dead branch detection, warnings for all sort of configs that won't work or will never be activated
-  return (
-    <Stack direction="column" gap={1.5}>
-      <div
-        className={styles.policyWrapper(hasFocus)}
-        data-testid={isDefaultPolicy ? 'am-root-route-container' : 'am-route-container'}
-      >
-        {/* continueMatching and showMatchesAllLabelsWarning are mutually exclusive so the icons can't overlap */}
-        {continueMatching && <ContinueMatchingIndicator />}
-        {showMatchesAllLabelsWarning && <AllMatchesIndicator />}
-        <div className={styles.policyItemWrapper}>
-          <Stack direction="column" gap={1}>
-            {/* Matchers and actions */}
-            <div>
-              <Stack direction="row" alignItems="center" gap={1}>
-                {isDefaultPolicy ? (
-                  <DefaultPolicyIndicator />
-                ) : hasMatchers ? (
-                  <Matchers matchers={matchers ?? []} />
-                ) : (
-                  <span className={styles.metadata}>No matchers</span>
-                )}
-                <Spacer />
-                {/* TODO maybe we should move errors to the gutter instead? */}
-                {errors.length > 0 && <Errors errors={errors} />}
-                {provisioned && <ProvisioningBadge />}
-                {!readOnly && (
+  return showCurrentPolicy ? (
```

Comment:
I'm a bit worried about this part - The conditionally rendered component is almost 200 lines long - so there is no way to see the second part of this condition
Should we try to extract smaller components for this? @gillesdemey

Reply:
maybe we could do an early `return null `instead of the conditional rendering? That would be more clear.

Reply:
I'm not too concerned about large files but definitely +1 for splitting up the logic in multiple components

Reply:
I added a commit splitting this component.